### ISSUE TITLE
How to finetune on a dataset perfectly? And how to evaluate the fine-tuned model?

### ISSUE THREAD

**Issue by ankurbhatia24** [facebookresearch/ParlAI#3761]
- I have finetuned the blender bot 90M model on my own dataset prepared in the format as mentioned https://parl.ai/docs/tutorial_task.html

- I have my dataset (total 300 lines) as a question and answer in each line and treated each question-answer pair as an episode as shown below:

text:question     labels:answer    episode_done:True
text:question     labels:answer    episode_done:True
..
..

- I want my model to fully learn this thing i.e. learn the answer to the question asked.

- I have fine-tuned the model using the following command: 

`python parlai/scripts/train_model.py  -t fromfile:parlaiformat --fromfile_datapath "/home/azureuser/ankur/aadhar.txt" -m transformer/generator --init-model zoo:tutorial_transformer_generator/model --dict-file zoo:tutorial_transformer_generator/model.dict --embedding-size 512 --n-layers 8 --ffn-size 2048 --dropout 0.1 --n-heads 16 --learn-positional-embeddings True --n-positions 512 --variant xlm --activation gelu --skip-generation True --fp16 True --text-truncate 512 --label-truncate 128 --dict-tokenizer bpe --dict-lower True -lr 1e-06 --optimizer adamax --lr-scheduler reduceonplateau --gradient-clip 0.1 -veps 0.25 --betas 0.9,0.999 --update-freq 1 --attention-dropout 0.0 --relu-dropout 0.0 --skip-generation True -vp 15 -stim 60 -vme 20000 -bs 32 -vmt ppl -vmm min --save-after-valid True --model-file /home/azureuser/ankur/testmodel/model_90M/test_train_90M_V8 --gpu 0 --tensorboard_log True --tensorboard_logdir /home/azureuser/ankur/tensorboard/test_train_90M_V8`

- The model trains for infinite time and I stopped the model training after 5k steps.
- I ran the model using the following command to check whether it has learnt or not:
`python parlai/scripts/interactive.py -t fromfile:parlaiformat --fromfile_datapath "/home/azureuser/ankur/aadhar.txt" -mf /home/azureuser/ankur/testmodel/model_90M/test_train_90M_V8 --init-model /home/azureuser/ankur/testmodel/model_90M/test_train_90M_V8.checkpoint`
- but the model showed random results and not the exact results. 
- Also the model showed the output text with fewer words/sentences than the actual labels were given in the training data. (What should I change to get longer output results?)
- Also, I want to know whether internally does the algo truncate the long sentences labels or training or something? If so, what should I do to increase the sentence length.

- Further, I evaluated the model using the command: 
`parlai display_model -t fromfile:parlaiformat --fromfile_datapath "/home/azureuser/ankur/aadhar.txt" -mf ../testmodel/model_90M/test_train_90M_V8`
but all the model outputs were: 'No response' (Why this is happening) (Is my model not trained enough)

- What all improvements or changes should I do?

**Comment by stephenroller** [comment]
For the last bit, display_model got no response because you need to set `--skip-generation false`. This is an annoying thing.

As far as why it didn't learn perfectly, it's hard to tell from your description. But your learning rate is extremely low, and that it didn't stop suggests it was still improving
